src/Temporalio/Runtime/OpenTelemetryOptions.cs

```diff
@@ -0,0 +1,50 @@
+using System;
+using System.Collections.Generic;
+
+namespace Temporalio.Runtime
+{
+    /// <summary>
+    /// OpenTelemetry tracing/metric collector options. <see cref="OpenTelemetryOptions.Url" /> is
+    /// required.
+    /// </summary>
+    public class OpenTelemetryOptions : ICloneable
+    {
+        /// <summary>
+        /// URL for the OpenTelemetry collector.
+        /// </summary>
+        public string? Url { get; set; }
+
+        /// <summary>
+        /// Headers to include in OpenTelemetry calls.
+        /// </summary>
+        public IEnumerable<KeyValuePair<string, string>>? Headers { get; set; }
+
+        /// <summary>
+        /// How frequently in metrics should be exported.
+        /// </summary>
+        public TimeSpan? MetricPeriodicity { get; set; }
```

Comment:
I called this `MetricsExportInterval` in TS because I figured most users would have to look up what `Periodicity` is.
Up to you if you if you want to change.

Reply:
:+1: Will change on next PR